c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1612,6 +1646,12 @@ class THCCachingAllocator {
         device,
         ": did you call init?");
     Block* block = device_allocator[device]->malloc(device, size, stream);
+    memory_tracker.append_alloc_free_event(
```

Comment:
since this is after allocating the block, it won't capture the last block that caused the OOM. To capture that, you can  use a invalid ptr as the initial value and update it after the block is created. See D38065728

Reply:
I had this in a previous commit but I removed it because I though when an OOM occurs, the program will stop and the allocate/free sequence vector will be destroyed? But I am not sure. I was thinking that alternatively in M2 we could save the allocate/free sequence in a file and the API reads from that file? Please let me know what you think. Thank you.

Reply:
> I had this in a previous commit but I removed it because I though when an OOM occurs, the program will stop and the allocate/free sequence vector will be destroyed?

No, the Python side will through an OOM `RuntimeError`, and we can capture that to avoid process crash. As long as the process is still alive, the per-process singleton `CUDACachingAllocator` and its states should be still available .

Reply:
Done. Thank you